hubploy/auth.py

```diff
@@ -127,6 +132,17 @@ def registry_auth_azure(deployment, resource_group, registry, auth_file):
         '--name', registry
     ])
 
+def registry_auth_docker(deployment, auth_file):
+    # parse auth file
+    auth_file_path = os.path.join('deployments', deployment, 'secrets', auth_file)
+    with open(auth_file_path) as f:
+        auth = yaml.load(f)
+
+    registry = auth.get('registry', 'https://index.docker.io/v1/')
+
+    client = docker.DockerClient()
+    client.login(username=auth['username'], password=auth['password'],
+        registry=registry)
```

Comment:
Does this actually log in for other calls on this machine, or just for this client object? I thought it was just for this particular client object...

Reply:
Good question, I don't know. I initially used subprocess with the docker cli to do a login but then changed it to the library when I saw you use it in the other files. I'll take a look.

Reply:
Looks like login() checks for and uses pre-existing login token in ~/.docker/config.json, otherwise it does the login and saves the results.
https://github.com/docker/docker-py/blob/master/docker/api/daemon.py#L97

Reply:
Digging through, I see it calls https://github.com/docker/docker-py/blob/master/docker/auth.py#L316, which doesn't persist that in disk. This makes me feel it is just for this client object.